### ISSUE TITLE
Add DropDownButtonRevealStyle

### ISSUE THREAD

**Issue by lhak** [Microsoft/microsoft-ui-xaml#70]
# Proposal: Add DropDownButtonRevealStyle

## Summary

Add a reveal style for the DropDownButton control.

## Rationale

There are currently reveal styles for many controls of the platform. However, there is one missing for the DropDownButton control. It would be nice if such a style could be added to WinUI.

## Functional Requirements

The style should have a similar design to the current ButtonRevealStyle/AppBarButtonRevealStyle styles.

**Comment by jevansaks** [comment]
Yup, totally agree. This seems to be an oversight. @chigy @kikisaints could you see about getting a comp for this?

I don't know when our team would be able to get to this, but we would welcome a contribution for this.

**Comment by kikisaints** [comment]
Actually, since DropDownButton's template is based on button, simply doing this:
```XAML
<DropDownButton Content="Email" Style="{ThemeResource ButtonRevealStyle}"/>
``` 
Will give you full Reveal on the DropDownButton.

However, for clarity's sake, I'd suggest adding something like this to generic.xaml:
```XAML
<Style TargetType="DropDownButton" x:Key="DropDownButtonRevealStyle" BasedOn="{StaticResource ButtonRevealStyle}"/>
```
@jevansaks I don't believe a comp is necessary, as testing out the ButtonRevealStyle on the DropDownButton gives the desired results and can be viewed today - but let me know if you feel otherwise.

**Comment by jevansaks** [comment]
I forgot to call out that I think we should also do a SplitButton reveal style along with this. 

For DropDownButton, yes, the fix may be simple -- but it still needs to be done. :)

**Comment by kikisaints** [comment]
For SplitButton, I was imagining something like this:
![splitbutton_reveal](https://user-images.githubusercontent.com/6964369/49903252-6cef6300-fe1b-11e8-950d-8d6d3e067074.gif)

**Comment by lhak** [comment]
I tried using ButtonRevealStyle, but this does not show the dropdown indicator.

**Comment by kikisaints** [comment]
Ah! So you are correct, @lhak. For your immediate remedy then, you can do this:

```XAML
<DropDownButton Style="{ThemeResource ButtonRevealStyle}">
    <DropDownButton.Content>
        <StackPanel Orientation="Horizontal">
            <TextBlock
            Text="Email"
            VerticalAlignment="Center"/>
            
            <TextBlock
            x:Name="ChevronTextBlock"
            Grid.Column="1"
            FontFamily="Segoe MDL2 Assets"
            FontSize="12"
            Text="&#xE70D;"
            VerticalAlignment="Center"
            Margin="6,0,0,0"
            AutomationProperties.AccessibilityView="Raw"/>
        </StackPanel>
    </DropDownButton.Content>
    
    <DropDownButton.Flyout>
        <MenuFlyout Placement="Bottom">
            <MenuFlyoutItem Text="Send"/>
            <MenuFlyoutItem Text="Reply"/>
            <MenuFlyoutItem Text="Reply All"/>
        </MenuFlyout>
    </DropDownButton.Flyout>
</DropDownButton>
```

But @jevansaks, here's a prototype comp for DropDownButtonStyle:
![dropdownbutton_revealgif](https://user-images.githubusercontent.com/6964369/49955615-17b55f00-feb8-11e8-9416-043f184dae57.gif)

**Comment by lhak** [comment]
Thank you for the suggestion. I actually need to put this button into a commandbar so I have used some similar code in a retemplated AppBarButton. Still, an official AppBarDropDownButton control would be great.

**Comment by lhak** [comment]
It seems that the added style is incomplete and lacks the dropdown indicator. It also does not show up in the intellisense box.

**Comment by lhak** [comment]
I tested the updated package but I still see some issues (the first two might be limitations of Visual Studio):

- The intellisense context menu in Visual Studio does not offer this as a style for the DropDownButton
- Jumping to the style definition with F12 does not work
- When the control is disabled the text of the button is grayed out, but the dropdown symbol is not

**Comment by jevansaks** [comment]
Thanks! Yes, the first two are limitations of visual studio. Could you file those via the "send feedback" in VS?

The third one seems like a bug. @chingucoding @kaiguo can you take a look?

**Comment by chingucoding** [comment]
I will look into the third bug.
